fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/F-PLN.tsx

```diff
@@ -0,0 +1,1068 @@
+            }
+        }, true));
+
+        this.subs.push(this.props.displayEfobAndWind.sub(() => {
+            const data = this.props.data.get();
+            if (isWaypoint(data)) {
+                this.renderSpdAltEfobWind(data);
+            }
+        }, true));
+
+        this.subs.push(this.props.data.sub((data) => this.onNewData(data), true));
+
+        this.subs.push(this.props.selectedForRevision.sub((val) => {
+            if (val === true) {
+                this.identRef.getOrDefault().classList.add('selected');
+            } else {
+                this.identRef.getOrDefault().classList.remove('selected');
+            }
+        }));
+
+        this.subs.push(this.props.revisionsMenuIsOpened.sub((val) => {
+            if (val === false) {
+                this.props.selectedForRevision.set(false);
+                this.identRef.getOrDefault().classList.remove('selected');
+            }
+        }));
+
+        this.identRef.instance.addEventListener('click', () => {
+            this.props.openRevisionsMenuCallback();
+            this.props.selectedForRevision.set(true);
+        });
+    }
+
+    private onNewData(data: FplnLineDisplayData): void {
+        if (isWaypoint(data)) {
+            if (this.currentlyRenderedType !== FplnLineType.Waypoint) {
+                while (this.topRef.getOrDefault().firstChild) {
+                    this.topRef.getOrDefault().removeChild(this.topRef.getOrDefault().firstChild);
+                }
+                FSComponent.render(this.renderWaypointLine(), this.topRef.getOrDefault());
+                this.currentlyRenderedType = FplnLineType.Special;
+            }
+
+            if (this.props.lineColor.get() === FplnLineColor.Active) {
+                this.lineRef.getOrDefault().classList.remove('MFDFplnLineSecondary');
+                this.lineRef.getOrDefault().classList.remove('MFDFplnLineTemporary');
+            } else if (this.props.lineColor.get() === FplnLineColor.Secondary) {
+                this.lineRef.getOrDefault().classList.add('MFDFplnLineSecondary');
+                this.lineRef.getOrDefault().classList.remove('MFDFplnLineTemporary');
+            } else if (this.props.lineColor.get() === FplnLineColor.Temporary) {
+                this.lineRef.getOrDefault().classList.add('MFDFplnLineTemporary');
+                this.lineRef.getOrDefault().classList.remove('MFDFplnLineSecondary');
+            }
+
+            // TODO: Hold/turn direction
+            if (data.overfly === true) {
+                this.identRef.getOrDefault().innerHTML = `<span>${data.ident}<span style="font-size: 24px; vertical-align: baseline;">@</span></span>`;
+            } else {
+                this.identRef.getOrDefault().innerText = data.ident;
+            }
+
+            // TODO: Hold/turn direction
+            if (this.annotationRef.getOrDefault()) {
```

Comment:
**suggestion:** you can use shorter syntax: `this.annotationRef.getOrDefault()?.innerText = data.annotation;` to avoid needing to wrap it in an if-statement

Reply:
Doesn't work: ```The left-hand side of an assignment expression may not be an optional property access.```
I need to guard it with IFs...